syz-verifier/verifier.go

```diff
@@ -88,17 +88,20 @@ func Verify(res []*Result, prog *prog.Prog, s *Stats) *ResultReport {
 	pool0 := res[0].Pool
 	for _, cr := range rr.Reports {
 		cs := s.Calls[cr.Call]
+
 		for _, state := range cr.States {
 			// For each CallReport verify the ReturnStates from all the pools
 			// that executed the program are the same
 			if state0 := cr.States[pool0]; state0 != state {
 				cr.Mismatch = true
 				send = true
 
-				s.TotalMismatches++
-				cs.Mismatches++
-				cs.States[state] = true
-				cs.States[state0] = true
+				if _, ok := cs.States[state]; !ok {
+					cs.Mismatches++
```

Comment:
Do we still count total number of mismatches somewhere else?
I think total number counted unconditionally is a very important number, e.g. if 99% of programs result in the same mismatch.

Reply:
Modified to only increment mismatches once per `CallReport`.